app/controllers/api/songs_controller.rb

```diff
@@ -0,0 +1,102 @@
+class API::SongsController < API::APIController
+
+  before_action :require_edit_privileges, only: [:create, :update]
+  before_action :require_delete_privileges, only: [:destroy]
+
+  def show
+    song = Song.find_by(id: params[:id])
+    if song
```

Comment:
I'm tempted to do if (song = Song.find_by) but I got a warning in my editor that said "did you mean to do ==?`. But it clearly works. Is assignment in a conditional frowned upon? What's idiomatic ruby?

Reply:
I would do
```
song = Song.find_by(id: params[:id])
return head(:not_found) unless song
case params[:format]
...
```